src/deps/libuwsockets.cpp

```diff
@@ -1577,4 +1576,29 @@ extern "C"
     s->context->loop->data.last_write_failed = 1;
     us_poll_change(&s->p, s->context->loop, LIBUS_SOCKET_READABLE | LIBUS_SOCKET_WRITABLE);
   }
+
+  uint64_t uws_res_get_remote_address_info(uws_res_t *res, const char **dest, int *port, bool *is_ipv6)
+  {
+    // This function is manual inlining + modification of
+    //      us_socket_remote_address
+    //      AsyncSocket::getRemoteAddress
+    // To get { ip, port, is_ipv6 } for Bun.serve().requestIP()
+    //      AsyncSocket::addressAsText
+    static thread_local char b[64];
+    auto length = us_get_remote_address_info(b, (us_socket_t *)res, dest, port, (int*)is_ipv6);
+
+    if (length == 4) {
+        length = sprintf(b, "%u.%u.%u.%u", b[0], b[1], b[2], b[3]);
```

Comment:
we should use `snprintf` so that the length is always checked and there's no risk of buffer overflow

Reply:
this code is adapted from another part of uSockets

if b[0] is interpreted as a u8, then the max length of this string is 15

the longest ipv6 string is supposed to be 39 characters long, though i'm not certain if the sprintf would respect that bound.

the given `b` is always 64 bytes long, i don't think a length check is neccecary, but we can do it just to ensure these bounds.